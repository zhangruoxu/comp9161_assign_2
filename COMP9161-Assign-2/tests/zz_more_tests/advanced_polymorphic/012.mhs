main = letfun f x = 
  (let m = letfun h m = let z = x; in x; in m, x+1) ;



