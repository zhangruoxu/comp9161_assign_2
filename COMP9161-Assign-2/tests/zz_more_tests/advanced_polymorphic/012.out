main :: forall g. (Int -> (g -> Int) * Int) = (letfun f :: (Int -> (g -> Int) * Int) x = (let m :: forall e. (e -> Int) = (letfun h :: (e -> Int) m = let z :: Int = x;
                                                                                                                                                      in x);
                                                                                          in m, (+) x 1));
